numba/targets/arrayobj.py

```diff
@@ -2790,11 +2790,10 @@ def make_array_flatiter(context, builder, arrty, arr):
     flatitercls = make_array_flat_cls(types.NumpyFlatType(arrty))
     flatiter = flatitercls(context, builder)
 
-    arrayptr = cgutils.alloca_once_value(builder, arr)
```

Comment:
No need separate stack allocation anymore.  It is allocated as part of the iterator structure.